lib/stack_master/template_compiler.rb

```diff
@@ -1,8 +1,11 @@
 module StackMaster
   class TemplateCompiler
+    TemplateCompilationFailed = Class.new(RuntimeError)
 
     def self.compile(config, template_file_path)
       template_compiler_for_file(template_file_path, config).compile(template_file_path)
+    rescue StandardError
```

Comment:
Can we be more explicit here?

I think it would be preferable if each template compiler handled the relevant exceptions and re-raised them as StackMaster exceptions.